src/libraries/System.Runtime/tests/System/Type/TypePropertyTests.cs

```diff
@@ -672,7 +672,8 @@ public class IntPointerTests : TypePropertyTestBase
     {
         public override Type CreateType() => typeof(int).MakePointerType();
 
-        public override TypeAttributes Attributes => TypeAttributes.Class;
+        public override TypeAttributes Attributes => PlatformDetection.IsMonoRuntime ? TypeAttributes.Public :
```

Comment:
Can we fix Mono and/or CoreCLR to have identical behavior here?

Reply:
I saw several existing tests having conditional behavior so i thought difference is expected
https://github.com/dotnet/runtime/blob/11076ec2edbcd1e0972c4811299049cf7be19c83/src/libraries/System.Runtime/tests/System/Type/TypePropertyTests.cs#L538-L540
https://github.com/dotnet/runtime/blob/11076ec2edbcd1e0972c4811299049cf7be19c83/src/libraries/System.Runtime/tests/System/Type/TypePropertyTests.cs#L1096-L1098
Also mono and CLR implementations looks quite different, if we want same behavior i beleive that needs quite a lot code/test change needed.

Reply:
For the `Type.IsPublic` API side both has identical behavior here, but CLR produce more attributes other than `TypeAttributes.Public`, please let me know if we want same behavior and what/which behavior is expected

Reply:
> I saw several existing tests having conditional behavior so i thought difference is expected

The TypeAttributes flags are mix of flags that describe public surface (e.g. TypeAttributes.Public/NonPublic) and flags that describe internal implementation details (e.g. TypeAttributes.SequentialLayout/AutoLayout). The existing Mono conditions in these tests are for flags that describe internal implementation details because the internal implementation is actually different. That's ok - we do not guarantee that the internal implementation details are same between all runtimes.

Reply:
I am starting to think that we may want to unconditionally return TypeAttributes.Public for pointers and byrefs instead of inheriting it from element type, similar to what we are doing for generic variables.

The type visibility encoded in TypeAttribute flags is a local property of the given constituent. It does not says whether the composed type can be accessed publicly. For example, `typeof(List<MyPrivateType>).IsPublic` is going to be true since `List<T>` is a public type, even though this type is not actually publicly accessible.

Note that we also have `Type.IsVisible` property that checks local type visibility of all constituents and returns true only if all constituents are public.

Reply:
Returning true for `typeof(InternalStruct*).IsPublic` looks a bit odd to me 
```cs
    Console.WriteLine(typeof(PublicStruct).IsPublic); // true
    Console.WriteLine(typeof(PublicStruct*).IsPublic); // true
    Console.WriteLine(typeof(PublicStruct*).IsVisible); // true
    Console.WriteLine(typeof(InternalStruct).IsPublic); // false
    Console.WriteLine(typeof(InternalStruct*).IsPublic); // true
    Console.WriteLine(typeof(InternalStruct*).IsVisible); // false
```

Reply:
I think we should be treating generic types and parametrized type the same way. They are very similar and partially redundant concepts. If we have had generics in the first .NET release (.NET Framework 1.0), parametrized types would not possibly exist at all, and pointers or arrays would just be special generic types.

So I think the output of this:
```
Console.WriteLine(typeof(List<InternalStruct>).IsPublic);
Console.WriteLine(typeof(List<InternalStruct>).IsVisible);
should be identical to the output of this:
```
Console.WriteLine(typeof(InternalStruct*).IsPublic);
Console.WriteLine(typeof(InternalStruct*).IsVisible);
```